src/ts/settlement.ts

```diff
@@ -160,7 +180,10 @@ export const FLAG_MASKS = {
 
 export type FlagKey = keyof typeof FLAG_MASKS;
 export type FlagOptions<K extends FlagKey> = typeof FLAG_MASKS[K]["options"];
-export type FlagValue<K extends FlagKey> = FlagOptions<K>[number];
+export type FlagValue<K extends FlagKey> = Exclude<
+  FlagOptions<K>[number],
+  undefined
+>;
```

Comment:
This section requires extra attention

Reply:
This makes it so that undefined is not a valid value for the 2bit option?

Reply:
Correct. So anytime `undefined` appears in the `options` list in the `FLAG_MASKS`, they act as a "unused value".

For example, with:
```ts
type FlagValue<K extends FlagKey> = FlagOptions<K>[number];
let sellTokenBalance: FlagValue<"sellTokenBalance">; // OrderBalance.ERC20 | undefined | OrderBalance.EXTERNAL | OrderBalance.INTERNAL;
```

However, if we `Exclude<..., undefined>`, then:
```ts
type FlagValue<K extends FlagKey> = Exclude<FlagOptions<K>[number], undefined>;
let sellTokenBalance: FlagValue<"sellTokenBalance">; // OrderBalance.ERC20 | OrderBalance.EXTERNAL | OrderBalance.INTERNAL;
```